feeds/backend.php

```diff
@@ -4,22 +4,43 @@
 include_once($relPath.'pg.inc');
 
 $content = get_enumerated_param($_GET, 'content', 'posted', ['posted', 'postprocessing', 'proofing', 'smoothreading']); // Which feed the user wants
-// Time in seconds for how often the feeds get refreshed
-// Disable delay if we are on the test server
-$refreshDelay = $testing ? 0 : 30 * 60;
-$refreshAge = time() - $refreshDelay; // How long ago $refreshDelay was in UNIX time
-$limit = 20; // Number of rows we query from the table, number of items in RSS feed
+$rssfeed = generate_rss_feed($content, $site_name, $code_url, $charset, $site_manager_email_addr);
 
-// Determine if we should display a 0.91 compliant RSS feed or our own feed
-$intlang = get_desired_language();
-if (isset($_GET['type'])) {
-    $xmlfile = "$xmlfeeds_dir/{$content}_rss.$intlang.xml";
-} else {
-    $xmlfile = "$xmlfeeds_dir/{$content}.$intlang.xml";
-}
+// Let the browser cache it for $cache_duration seconds
+$cache_duration = 30 * 60;
+$now = time();
+header("Content-Type: text/xml; charset=$charset");
+header("Expires: " . gmdate("D, d M Y H:i:s", $now + $cache_duration) . " GMT");
+header("Last-Modified: " . gmdate("D, d M Y H:i:s", $now) . " GMT");
+header("Cache-Control: max-age=$cache_duration, public, must-revalidate");
+
+echo $rssfeed;
+
+/**
+ * Generate an RSS feed for a category of eBooks
+ *
+ * @param string $content
+ *   The category for the feed: posted, postprocessing, proofing, or smoothreading
+ *
+ * @param string $site_name
+ *   The name of the site serving the feed
+ *
+ * @param string $code_url
+ *   The base URL for the site
+ *
+ * @param string $charset
+ *   The character set of the feed, such as "UTF-8"
+ *
+ * @param string $site_manager_email_addr
+ *   Contact info for the feed <webMaster> element
+ *
+ * @return string
+ *   The XML document for the RSS feed
+ */
+function generate_rss_feed($content, $site_name, $code_url, $charset, $site_manager_email_addr)
```

Comment:
Please add type definitions here:
```suggestion
function generate_rss_feed(string $content, string $site_name, string $code_url, string $site_manager_email_addr): string
```

Reply:
Done. Is this a new practice? I didn't see it in [DP Code Best Practices](https://www.pgdp.net/wiki/DP_Code_Best_Practices).

Reply:
It is a new practice. I'll get the best practices code updated - thanks for calling it out.

Reply:
Added this (brief) section on typing here: https://www.pgdp.net/wiki/DP_Code_Best_Practices#Typing